addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/handler/MySensorsBridgeHandler.java

```diff
@@ -0,0 +1,238 @@
+
+        myGateway = new MySensorsGateway(loadCacheFile());
+
+        if (myGateway.setup(openhabToMySensorsGatewayConfig(myBridgeConfiguration, getThing().getThingTypeUID()))) {
+            myGateway.startup();
+
+            myGateway.addEventListener(this);
+
+            registerDeviceDiscoveryService();
+            // reloadSensors();
+
+            logger.debug("Initialization of the MySensors bridge DONE!");
+        } else {
+            logger.error("Failed to initialize MySensors bridge");
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing of the MySensors bridge");
+
+        unregisterDeviceDiscoveryService();
+
+        if (myGateway != null) {
+            myGateway.removeEventListener(this);
+            myGateway.shutdown();
+        }
+
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+    }
+
+    /**
+     * Getter for the configuration of the bridge.
+     *
+     * @return Configuration of the MySensors bridge.
+     */
+    public MySensorsBridgeConfiguration getBridgeConfiguration() {
+        return myBridgeConfiguration;
+    }
+
+    /**
+     * Getter for the connection to the MySensors bridge / gateway.
+     * Used for receiving (register handler) and sending of messages.
+     *
+     * @return Connection to the MySensors bridge / gateway.
+     */
+    public MySensorsGateway getMySensorsGateway() {
+        return myGateway;
+    }
+
+    @Override
+    public void connectionStatusUpdate(MySensorsAbstractConnection connection, boolean connected) throws Exception {
+        if (connected) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+
+        updateCacheFile();
+    }
+
+    @Override
+    public void nodeIdReservationDone(Integer reservedId) throws Exception {
+        updateCacheFile();
+    }
+
+    @Override
+    public void newNodeDiscovered(MySensorsNode node, MySensorsChild child) throws Exception {
+        updateCacheFile();
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.debug("Configuation update for bridge: {}", configurationParameters);
+        super.handleConfigurationUpdate(configurationParameters);
+    }
+
+    private void updateCacheFile() {
+
+        List<Integer> givenIds = myGateway.getGivenIds();
+
+        cacheFactory.writeCache(MySensorsCacheFactory.GIVEN_IDS_CACHE_FILE, givenIds.toArray(new Integer[] {}),
+                Integer[].class);
+    }
+
+    private Map<Integer, MySensorsNode> loadCacheFile() {
+        Map<Integer, MySensorsNode> nodes = new HashMap<Integer, MySensorsNode>();
+
+        List<Integer> givenIds = cacheFactory.readCache(MySensorsCacheFactory.GIVEN_IDS_CACHE_FILE,
+                new ArrayList<Integer>(), new TypeToken<ArrayList<Integer>>() {
```

Comment:
What are you achieving with `new TypeToken<ArrayList<Integer>>() {}.getType()` creating an anonymous inner class instance and calling some method...